src/main/java/org/desarrolladorslp/technovation/services/impl/BatchServiceImpl.java

```diff
@@ -50,6 +51,20 @@ public Batch save(Batch batch) {
         return batchRepository.findByProgram(program);
     }
 
+    @Override
+    @Transactional
+    public Optional<Batch> delete(UUID id){
+        Optional<Batch> optionalBatch = batchRepository.findById(id);
```

Comment:
Do we really need to return the removed batch? .. I think you can only have a `void` method here and return only a 200 in the controller.

If you prefer to return the `Batch` I recommend to return a `Batch` instead of an `Optional` and throw a `NotFoundException` from inside the method if the batch does not exist

Reply:
> Why are you keeping the `Optional<Batch>`?

nvm ... I saw you need that batch to call `delete` method